lib/migrate/extractTextPlugin/extractTextPlugin.js

```diff
@@ -1,58 +1,111 @@
-const utils = require("../../utils/ast-utils");
+const {
+	addRequire,
+	createPluginByName,
+	removePluginByName,
+	getLoaderOptionValue,
+	getPluginOptionValue,
+	isType,
+	pathToArray,
+	removeRequire,
+	safeTraverse
+} = require("../../utils/ast-utils");
 
 /**
  *
- * Check whether `node` is the invocation of the plugin denoted by `pluginName`
- *
- * @param {Object} j - jscodeshift top-level import
- * @param {Node} node - ast node to check
- * @param {String} pluginName - name of the plugin
- * @returns {Boolean} isPluginInvocation - whether `node` is the invocation of the plugin denoted by `pluginName`
- */
-
-function findInvocation(j, node, pluginName) {
-	let found = false;
-	found =
-		j(node)
-			.find(j.MemberExpression)
-			.filter(p => p.get("object").value.name === pluginName)
-			.size() > 0;
-	return found;
-}
-
-/**
- *
- * Transform for ExtractTextPlugin arguments. Consolidates arguments into single options object.
+ * Transform for ExtractTextPlugin. Replaces it with MiniCssExtractPlugin and translates
+ * configuration where possible.
  *
  * @param {Object} j - jscodeshift top-level import
  * @param {Node} ast - jscodeshift ast to transform
  * @returns {Node} ast - jscodeshift ast
  */
-
 module.exports = function(j, ast) {
-	const changeArguments = function(p) {
-		const args = p.value.arguments;
-
-		const literalArgs = args.filter(p => utils.isType(p, "Literal"));
-		if (literalArgs && literalArgs.length > 1) {
-			const newArgs = j.objectExpression(
-				literalArgs.map((p, index) =>
-					utils.createProperty(j, index === 0 ? "fallback" : "use", p.value)
-				)
-			);
-			p.value.arguments = [newArgs];
+	const devLoaders = [];
```

Comment:
Can you abstract all this into more functions that can be reused, like find value "filename" for a given prop

Reply:
I'm not sure I follow; we already abstracted functionality into simple, single-purpose functions like `getPluginOptionValue` and `getPluginLoaderValue`. If we start getting even more abstract than this, we risk targeting nodes that aren't actually the nodes we care about. For example, someone could define `filename` completely outside of the webpack configuration object, so we need to be very careful.

The functions I already pulled out (`createPluginByName`, `removePluginByName`, `getLoaderOptionValue`, `getPluginOptionValue`, and `removeRequire`) are reusable and safe.